src/main/java/com/evilcorp/bloom/service/CategoryService.java

```diff
@@ -0,0 +1,85 @@
+package com.evilcorp.bloom.service;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.stereotype.Service;
+
+import com.evilcorp.bloom.dto.CategoryDto;
+import com.evilcorp.bloom.model.Category;
+import com.evilcorp.bloom.repo.CategoryRepo;
+import com.evilcorp.bloom.exception.NestedSubcategoriesException;
+import com.evilcorp.bloom.exception.NotFoundException;
+
+@Service
+public class CategoryService {
+
+  private final CategoryRepo categoryRepo;
+  private final String anyWhitespace;
+  private final String emptySpace;
+
+  public CategoryService(CategoryRepo categoryRepo) {
+    this.categoryRepo = categoryRepo;
+    this.anyWhitespace = "\\s+";
+    this.emptySpace = " ";
+
+  }
+
+  public String capitalize(String categoryName) {
+    /**
+     * Capitalizes every word in a given String.
```

Comment:
Javadoc should be made outisde of the method

Reply:
oops fixed